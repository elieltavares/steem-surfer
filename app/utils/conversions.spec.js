/* eslint-disable */
import { vestsToSp, spToVests, vestsToRshares } from './conversions';

describe('conversions.js', () => {
  it('(1) vestsToSp', () => {
    const vests = 350;
    const steemPerMVests = 495.05469644322403;

    expect(vestsToSp(vests, steemPerMVests)).toMatchSnapshot();
  });

  it('(2) spToVests', () => {
    const sp = 0.17326914375512842;
    const steemPerMVests = 495.05469644322403;

    expect(spToVests(sp, steemPerMVests)).toMatchSnapshot();
  });

  it('(3) vestsToRshares', () => {
    const vests = 173734.018323;
    const votingPower = 9800;
    const votePerc = 32 * 100;

    expect(vestsToRshares(vests, votingPower, votePerc)).toMatchSnapshot();
  });
});
