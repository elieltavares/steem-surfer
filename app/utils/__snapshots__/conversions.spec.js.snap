// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversions.js (1) vestsToSp 1`] = `0.17326914375512842`;

exports[`conversions.js (2) spToVests 1`] = `350.00000000000006`;

exports[`conversions.js (3) vestsToRshares 1`] = `1107033164.754156`;
